internal/pkg/cli/storage_init.go

```diff
@@ -737,14 +763,33 @@ func (o *initStorageOpts) newRDSTemplate() (*addon.RDSTemplate, error) {
 		return nil, err
 	}
 
-	return addon.NewRDSTemplate(addon.RDSProps{
-		ClusterName:    o.storageName,
-		Engine:         engine,
-		InitialDBName:  o.rdsInitialDBName,
-		ParameterGroup: o.rdsParameterGroup,
-		Envs:           envs,
-		WorkloadType:   o.workloadType,
-	}), nil
+	var version string
+	switch o.auroraServerlessVersion {
+	case auroraServerlessVersionV1:
+		version = addon.AuroraServerlessVersionV1
+		return addon.NewServerlessV1Template(addon.RDSProps{
+			ClusterName:             o.storageName,
+			AuroraServerlessVersion: version,
+			Engine:                  engine,
+			InitialDBName:           o.rdsInitialDBName,
+			ParameterGroup:          o.rdsParameterGroup,
+			Envs:                    envs,
+			WorkloadType:            o.workloadType,
+		}), nil
+	case auroraServerlessVersionV2:
+		version = addon.AuroraServerlessVersionV2
+		return addon.NewServerlessV2Template(addon.RDSProps{
+			ClusterName:             o.storageName,
+			AuroraServerlessVersion: version,
+			Engine:                  engine,
+			InitialDBName:           o.rdsInitialDBName,
+			ParameterGroup:          o.rdsParameterGroup,
+			Envs:                    envs,
+			WorkloadType:            o.workloadType,
+		}), nil
+	default:
+		return nil, errors.New("unknown serverless version")
+	}
```

Comment:
To reduce duplication:
```go
	props := addon.RDSProps{
		ClusterName:             o.storageName,
		Engine:                  engine,
		InitialDBName:           o.rdsInitialDBName,
		ParameterGroup:          o.rdsParameterGroup,
		Envs:                    envs,
		WorkloadType:            o.workloadType,
	}
	switch v:= o.auroraServerlessVersion; v {
	case auroraServerlessVersionV1:
		return addon.NewServerlessV1Template(props), nil
	case auroraServerlessVersionV2:
		return addon.NewServerlessV2Template(props), nil
	default:
		return nil, fmt.Errorf("unknown Aurora serverless version %q", v)
	}
```
We shouldn't need to set the `AuroraServerlessVersion` as that's done by the constructor in the `addon` pkg

Reply:
Yes I noticed. I reduced duplication.